nir/ir.py

```diff
@@ -11,6 +12,15 @@
 # Shape is a dict mapping strings to shapes
 Shape = typing.Dict[str, np.ndarray]
 
+def _parse_shape_argument(x: Shape, key: str):
+    if isinstance(x, np.ndarray):
+        return {key: x}
+    elif isinstance(x, Sequence):
+        return {key: np.array(x)}
+    elif isinstance(x, dict):
+        return x
```

Comment:
perhaps here we can also assert that the dict has only one key which is the key passed into this function (i.e. "input")? in other places we assume that the only input key is "input", and this will break unless we enforce that here.

Reply:
That's a good point. I'll leave it for now, just to move forward here. In principle, I think these kinds of assertions are meaningful, but I'm cautious to add them because they can break people's code (maybe someone wants to violate that some day?) and they take a bit of runtime overhead. The last point is probably minimal. But still :)